pkg/java/opcodes.go

```diff
@@ -0,0 +1,266 @@
+	NoOperandOpcodes[0xbf] = true // athrow
+	NoOperandOpcodes[0xc2] = true // monitorenter
+	NoOperandOpcodes[0xc3] = true // monitorexit
+}
+
+var SingleOperandOpcodes = make([]bool, 0xff)
+
+func setSingleOperandOpcodes() {
+	SingleOperandOpcodes[0xbc] = true // newarray
+	SingleOperandOpcodes[0x10] = true // bipush
+	SingleOperandOpcodes[0x12] = true // ldc
+	SingleOperandOpcodes[0x15] = true // iload
+	SingleOperandOpcodes[0x16] = true // lload
+	SingleOperandOpcodes[0x17] = true // fload
+	SingleOperandOpcodes[0x18] = true // dload
+	SingleOperandOpcodes[0x19] = true // aload
+	SingleOperandOpcodes[0x36] = true // istore
+	SingleOperandOpcodes[0x37] = true // lstore
+	SingleOperandOpcodes[0x38] = true // fstore
+	SingleOperandOpcodes[0x39] = true // dstore
+	SingleOperandOpcodes[0x3a] = true // astore
+	SingleOperandOpcodes[0xa9] = true // ret
+}
+
+var DoubleOperandOpcodes = make([]bool, 0xff)
+
+func setDoubleOperandOpcodes() {
+	DoubleOperandOpcodes[0x99] = true // ifeq
+	DoubleOperandOpcodes[0x9a] = true // ifne
+	DoubleOperandOpcodes[0x9b] = true // iflt
+	DoubleOperandOpcodes[0x9c] = true // ifge
+	DoubleOperandOpcodes[0x9d] = true // ifgt
+	DoubleOperandOpcodes[0x9e] = true // ifle
+	DoubleOperandOpcodes[0x9f] = true // if_icmpeq
+	DoubleOperandOpcodes[0xa0] = true // if_icmpne
+	DoubleOperandOpcodes[0xa1] = true // if_icmplt
+	DoubleOperandOpcodes[0xa2] = true // if_icmpge
+	DoubleOperandOpcodes[0xa3] = true // if_icmpgt
+	DoubleOperandOpcodes[0xa4] = true // if_icmple
+	DoubleOperandOpcodes[0xa5] = true // if_acmpeq
+	DoubleOperandOpcodes[0xa6] = true // if_acmpne
+	DoubleOperandOpcodes[0xa7] = true // goto
+	DoubleOperandOpcodes[0xa8] = true // jsr
+	DoubleOperandOpcodes[0xc6] = true // ifnull
+	DoubleOperandOpcodes[0xc7] = true // ifnonnull
+	DoubleOperandOpcodes[0x11] = true // sipush
+	DoubleOperandOpcodes[0x84] = true // iinc
+	DoubleOperandOpcodes[0x13] = true // ldc_w
+	DoubleOperandOpcodes[0x14] = true // ldc2_w
+	DoubleOperandOpcodes[0xb2] = true // getstatic
+	DoubleOperandOpcodes[0xb3] = true // putstatic
+	DoubleOperandOpcodes[0xb4] = true // getfield
+	DoubleOperandOpcodes[0xb5] = true // putfield
+	DoubleOperandOpcodes[0xb6] = true // invokevirtual
+	DoubleOperandOpcodes[0xb7] = true // invokespecial
+	DoubleOperandOpcodes[0xb8] = true // invokestatic
+	DoubleOperandOpcodes[0xbb] = true // new
+	DoubleOperandOpcodes[0xbd] = true // anewarray
+	DoubleOperandOpcodes[0xc0] = true // checkcast
+	DoubleOperandOpcodes[0xc1] = true // instanceof
+}
+
+var QuadOperandOpcodes = make([]bool, 0xff)
+
+func setQuadOperandOpcodes() {
+	QuadOperandOpcodes[0xc8] = true // goto_w
+	QuadOperandOpcodes[0xc9] = true // jsr_w
+	QuadOperandOpcodes[0xba] = true // invokedynamic
+	QuadOperandOpcodes[0xb9] = true // invokeinterface
+}
+
+var TripleOperandOpcodes = []uint8{0xc5}
+var OtherOpcodes = []uint8{0xc4, 0xab, 0xaa, 0xfe, 0xff, 0xca}
+
+var OpcodesInitialised = false
+
+func OpcodeLookupTables() Opcodes {
+	if !OpcodesInitialised {
```

Comment:
Any reason to not initialise these everytime?
It would make the code a little simpler.

Reply:
This will eventually be called on every .class file found (at least with the most aggressive approach to scanning) so I assume that's a lot of thrash/updating memory needlessly. If the golang compiler is smarter than me (likely) it's possible this is premature and I can remove this.

Reply:
Discussed on VTC. Moving forwards with this implementation 👌